tests/js/session-tests.js

```diff
@@ -1041,6 +1051,114 @@ module.exports = {
                             resolve('Done');
                         }
                     });
+                    setTimeout(() => { reject() }, 10000);
+                }).catch(error => reject(error));
+            });
+        });
+    },
+
+    testStartStop() {
+        if(!isNodeProccess) {
+            return;
+        }
+
+        return Realm.Sync.User.register('http://localhost:9080', uuid(), 'password').then((u) => {
+            return new Promise((resolve, reject) => {
+                const config = {
+                    sync: {
+                        user: u,
+                        url: `realm://localhost:9080/~/${uuid()}`,
+                        fullSynchronization: true,
+                    }
+                };
+
+                const checks = {
+                    started: false,
+                    stopped: false,
+                    restarted: false,
+                }
+
+                Realm.open(config).then(realm => {
+                    const session = realm.syncSession;
+
+                    session.addConnectionNotification((newState, oldState) => {
+                        if (newState === Realm.Sync.ConnectionState.Connected && checks.started === false) { checks.started = true; session.stop(); }
```

Comment:
This is very convoluted to read. Any reason not to spread them on multiple lines?